expression/builtin_string.go

```diff
@@ -2959,18 +2959,42 @@ func (c *fromBase64FunctionClass) getFunction(ctx sessionctx.Context, args []Exp
 	}
 	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString)
 	bf.tp.Flen = mysql.MaxBlobWidth
+
+	valStr, _ := ctx.GetSessionVars().GetSystemVar(variable.MaxAllowedPacket)
+	maxAllowedPacket, err := strconv.ParseUint(valStr, 10, 64)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	types.SetBinChsClnFlag(bf.tp)
-	sig := &builtinFromBase64Sig{bf}
+	sig := &builtinFromBase64Sig{bf, maxAllowedPacket}
 	return sig, nil
 }
 
+// base64NeededDecodedLength return the base64 decoded string length.
+func base64NeededDecodedLength(n int) int {
+	// Returns -1 indicate the result will overflow.
+	if strconv.IntSize == 64 {
+		if n > 3074457345618258602 {
```

Comment:
Is this value equal to `math.MaxInt64`?

Reply:
No, it's equals to `0x2AAAAAAAAAAAAAAALL`, I copy it from mysql.
https://github.com/mysql/mysql-server/blob/5.7/mysys/base64.c#L69
I haven't figured out how it works, do you have any ideas?

Reply:
It is better to define a constant instead of using a magic number.

Reply:
It is `math.MaxInt64/3`

Reply:
how about:
```go
if strconv.IntSize == 64 && n > math.MaxInt64/3 {
	return -1
}
if strconv.IntSize == 32 && n > math.MaxInt32/3 {
	return -1
}
```

Reply:
Sure, Thanks a lot!! I will correct it.
I still confuse about why it is `math.MaxInt64/3`. Does that mean `base64_decode` function malloc 3 times of source string length during the process? It doesn't make sense.

Reply:
This check is to guarantee that the later `n*3` instruction, which is part of `n*3/4`, does overflow the max int64 or int32 value.

Reply:
Oh, I get it, thanks~

Reply:
😄